### DISCUSSION TITLE
Welcome to Data-Structures-and-Algorithms Discussions!

### DISCUSSION THREAD

**Discussion by Anupam0401** [Anupam0401/Data-Structures-and-Algorithms#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Amanuttam1192** [comment]
How come the TC of krushkal is O(E log V),rather than it having having the visited node in the glance

**Reply by Anupam0401** [reply]
Hello Aman,
In Kruskal's algorithm to find the minimum spanning tree, we are first sorting the edges based on their weights which amounts to O(E log E) time complexity. After this, we traverse the sorted edges one by one to find out if they belong to the same component or not which is done in O(E) time complexity so, the overall time complexity is O(E log E) or O(E log V).
Hope your query is answered :)
Thank you for asking here.

**Reply by Amanuttam1192** [reply]
Thanks for replying 
But what is the need of traversing each and every node if it can be done while visiting once.

**Reply by Amanuttam1192** [reply]
And how can we get the exact time complexity by just assuming the edges

**Reply by Anupam0401** [reply]
We are not traversing nodes here, we are traversing the edges if you go through the algorithm carefully.
I think you agree that we need to traverse each and every edge once to check if it is being reached from the spanning tree formed till then and accordingly add or don't add that edge in the spanning tree.
Also, we are not assuming the edges, the graph is an input to this algorithm i.e., all its nodes and edges are given priorly.

**Reply by Amanuttam1192** [reply]
ok ok may be thats enough for me to get and idea
thanks for your help

**Comment by Anupam0401** [comment]
You can find the pseudo-code for Kruskal's algorithm here:
```
KRUSKAL(G):
A = ∅
For each vertex v ∈ G.V:
    MAKE-SET(v)
For each edge (u, v) ∈ G.E ordered by increasing order by weight(u, v):
    if FIND-SET(u) ≠ FIND-SET(v):       
    A = A ∪ {(u, v)}
    UNION(u, v)
return A
```

**Reply by Amanuttam1192** [reply]
i got it thanks a lot

**Reply by Anupam0401** [reply]
happy to help  :)
